### PR TITLE
Add deploy functionality to packer-templates!

### PR DESCRIPTION
**Pull request by Codysseus**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -12,23 +12,31 @@ everything runs on whatever is labeled as 'master'
...
          error("Stopping because this build was not triggered on a PR's synchronize/opened/review_requested event which would have all the necessary information for making a build successful")
       }
 
+      if ( event == 'created' ){
+        comment = new JsonSlurper().parseText("${params.payload}")['comment']['body']
+        if ( comment != '!deploy' ){
+          currentBuild.result = 'ABORTED'
+          error("Stopping this build because the comment was not a recognized command.")
+        }
+      }
```
Comment (ramereth) [inline Jenkinsfile]: I'm trying to understand what this is trying to avoid. It looks like when we create a new PR, you want to make sure it doesn't have ``!deploy`` in it?

Edit: I read below in the comment that when a comment is created it uses the "created" event. Can you please comment it here too?

File: Jenkinsfile [hunk 2]
```diff
@@ -12,23 +12,31 @@ everything runs on whatever is labeled as 'master'
...
+
+      //clone the osl-jenkins on master branch to use the latest version of scripts
+      git url: 'https://github.com/osuosl-cookbooks/osl-jenkins', branch: 'master'
+
+      //write payload to a file
+      writeFile file: "/tmp/packer_pipeline_job_build_$BUILD_NUMBER", text: "$params.payload"
```
Comment (ramereth) [inline Jenkinsfile]: Why was this moved lower into the file?

File: Jenkinsfile [hunk 3]
```diff
@@ -348,6 +361,30 @@ def deploy_image_for_testing(arch) {
    templates = null
 }
 
+/* deploy_image_on_production(arch)
+
+   deploys qcow2 images on the various clusters for the given arch.
+   cluster credentials come from packer_pipeline_credentials.json
+   in alfred's home
+
+   TODO: failure while deploying any image is a soft failure
```
Comment (ramereth) [inline Jenkinsfile]: What's defined as a "soft" failure?

File: Jenkinsfile (hunk 2 above)
Comment (Codysseus) [reply Jenkinsfile]: I figured we didn't need to clone the PR or write the payload to a file if the jenkins job is just going to be aborted.

File: Jenkinsfile (hunk 3 above)
Comment (Codysseus) [reply Jenkinsfile]: I copied this from Samar's function. Soft failure probably means the other images are tested even if this image fails. Currently if an image fails all the tests stop.

**Comment by Codysseus** [comment]
@ramereth Are they any other changes that I should make?
